File: library-java/src/main/java/com/leeiidesu/loading/java/change/impl/SwitchLayoutHelperFactory.java
package com.leeiidesu.loading.java.change.impl;

import android.view.View;

import com.leeiidesu.loading.java.change.Mode;
import com.leeiidesu.loading.java.change.SwitchLayoutHelper;

/**
 * Created by leeiides on 2017/6/23.
 */

public class SwitchLayoutHelperFactory {


    public static SwitchLayoutHelper getSwitchLayoutHelper(Mode mode, View view) {
        switch (mode) {
            case REPLACE:
                return new ReplaceSwitchLayoutHelper(view);
            case COVER:
                return new CoverSwitchLayoutHelper(view);
        }
        return new ReplaceSwitchLayoutHelper(view);
    }

    public static SwitchLayoutHelper getSwitchLayoutHelperFromReplaceMode(View view) {
        return new ReplaceSwitchLayoutHelper(view);
    }

    public static SwitchLayoutHelper getSwitchLayoutHelperFromCoverMode(View view) {
        return new CoverSwitchLayoutHelper(view);
    }

}

File: library-kotlin/src/main/java/com/leeiidesu/loading/library/change/SwitchLayoutHelperFactory.kt
package com.leeiidesu.loading.library.change

import android.view.View
import com.leeiidesu.loading.library.change.impl.CoverSwitchLayoutHelper
import com.leeiidesu.loading.library.change.SwitchLayoutHelper
import com.leeiidesu.loading.library.change.impl.ReplaceSwitchLayoutHelper

/**
 * Created by leeiides on 2017/6/23.
 */
class SwitchLayoutHelperFactory {
    enum class Mode {
        REPLACE, COVER
    }

    companion object {
        fun getSwitchLayoutHelper(mode: Mode, originView: View): SwitchLayoutHelper {
            when (mode) {
                Mode.REPLACE -> return ReplaceSwitchLayoutHelper(originView)
                Mode.COVER -> return CoverSwitchLayoutHelper(originView)
            }
        }
    }
}

